Polyfills/XMLHttpRequest/Source/XMLHttpRequest.cpp

```diff
@@ -212,6 +266,9 @@ namespace Babylon::Polyfills::Internal
             throw Napi::Error::New(info.Env(), "XMLHttpRequest must be opened before it can be sent");
         }
 
+        std::string requestBody = info[0].IsString() ? info[0].As<Napi::String>().Utf8Value() : std::string();
+        m_request.SetRequestBody(requestBody);
```

Comment:
I think it would make more sense like this.
```suggestion
        if (info.Length() > 0)
        {
            if (!info[0].IsString())
            {
                throw Napi::Error:New(info.Env(), "Only strings are supported in XMLHttpRequest body");
            }

            std::string requestBody = info[0].As<Napi::String>().Utf8Value();
            m_request.SetRequestBody(requestBody);
        }
```

Reply:
Instead of doing `if (!info[0].IsString())` I did `if (!info[0].IsString() && !info[0].IsUndefined())`, to allow undefined arguments (based on this discussion, the tests now pass an undefined arguments into `xhr.send()`: https://github.com/BabylonJS/JsRuntimeHost/pull/12#discussion_r1127174569)

Reply:
Additionally, since `info[0]` can be undefined, I've reverted back to ` info[0].IsString() ? info[0].As<Napi::String>().Utf8Value() : std::string()` from `info[0].As<Napi::String>().Utf8Value();` otherwise windows throws an error.

Reply:
Yes, my code doesn't account for undefined or null.